### PR TITLE
ci: enforce conventional commits format on PR titles 

### PR DESCRIPTION
**Pull request by yogeshchoudhary147**
## Summary

- Adds a `lint-pr-title` local action to validate PR titles against the Conventional Commits specification using commitlint
- Adds `commitlint.config.mjs` at the repo root with inline rules (no external `extends` dependency)
- Adds `.github/workflows/lint-pr-title.yml` triggered on `pull_request` (works for forks on public repos without needing elevated permissions)
- Action runs on Node 24 with ESM modules

## Test plan

- [x] Open a PR with an invalid title (e.g. `my change`) and verify the check fails
- [x] Open a PR with a valid title (e.g. `fix: correct token expiry edge case`) and verify the check passes
- [ ] Verify the check runs on PRs from forks

### CODE REVIEW

File: .github/actions/lint-pr-title/src/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    if (!prTitle) {
+      core.setFailed('⛔️ Could not get pull request title. This action should only be run on a pull_request event.');
+      return;
+    }
+
+    core.info(`📝 Validating PR Title: "${prTitle}"`);
+
+    // 2. Load the commitlint configuration from the repository
+    const config = await load();
```
Comment (arpit-jn) [inline .github/actions/lint-pr-title/src/index.js]: `@commitlint/load` does not throw when no config file is discovered. Tt returns `{ rules: {} }`. With empty rules, `lint(prTitle, {})` returns `valid: true` for every title, while the action still logs "Loaded successfully." So if `commitlint.config.mjs` is ever renamed, moved, or the action's cwd shifts, enforcement is silently disabled, and all PRs pass, the worst failure mode for a guard rail. Could we add an assertion?

```
const config = await load();
if (!config.rules || Object.keys(config.rules).length === 0) {
  core.setFailed('⛔️ No commitlint rules loaded. Is commitlint.config.mjs present at the repo root?');
  return;
}
```

File: .github/workflows/lint-pr-title.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Checkout code
+        uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
+
+      - name: Setup Node
+        uses: actions/setup-node@48b55a011bda9f5d6aeb4c2d9c7362e8dae4041e # v6.4.0
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Install action dependencies
+        run: npm install
```
Comment (arpit-jn) [inline .github/workflows/lint-pr-title.yml]: On a `pull_request` event, the fork's head is checked out, so its `package.json` is attacker-controlled; `npm install` runs `postinstall`, etc. → arbitrary code on the runner. Blast radius is limited (read-only token, no secrets on fork PRs, but `--ignore-scripts` is free hardening and is fully compatible with your no-lockfile/float-to-patched design:

`run: npm install --ignore-scripts`

This is exactly the path your unchecked test-plan box ("runs on PRs from forks") exercises.

File: .github/workflows/lint-pr-title.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+name: Lint PR Title
+
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+
+permissions:
```
Comment (arpit-jn) [inline .github/workflows/lint-pr-title.yml:9]: The title is read from `github.context.payload` (the event), never via API, so `pull-requests: read` is unused, while `actions/checkout` wants `contents: read`. It works today only because anonymous read works on a public repo. (Dropping my earlier "repo goes private" justification, it's public and staying so.) Set:

```
permissions:
  contents: read
```

File: .github/actions/lint-pr-title/src/index.js (hunk 1 above)
Comment (yogeshchoudhary147) [reply .github/actions/lint-pr-title/src/index.js]: Fixed — passing explicit `file` to `load()` so it throws on missing config, plus a rules-length guard as a second layer.

File: .github/workflows/lint-pr-title.yml (hunk 2 above)
Comment (yogeshchoudhary147) [reply .github/workflows/lint-pr-title.yml]: Fixed — added `--ignore-scripts`; also noted the root `.npmrc` has `ignore-scripts=true` but doesn't apply here since the action has its own `package.json`.

File: .github/workflows/lint-pr-title.yml (hunk 3 above)
Comment (yogeshchoudhary147) [reply .github/workflows/lint-pr-title.yml:9]: Fixed — `pull-requests: read` → `contents: read`, consistent with every other workflow in this repo that only needs checkout.

**Review by arpit-jn — APPROVED** [review]
